lib/ecmascript.ts

```diff
@@ -3679,151 +3831,204 @@ export const ES = ObjectAssign({}, ES2020, {
-      case 'nanosecond':
-        quantity = quantity.multiply(1000).plus(nanosecond);
-    }
-    const nsPerUnit = unit === 'day' ? dayLengthNs : nsPerTimeUnit[unit];
-    const rounded = ES.RoundNumberToIncrement(quantity, nsPerUnit * increment, roundingMode);
-    const result = rounded.divide(nsPerUnit).toJSNumber();
-    switch (unit) {
-      case 'day':
-        return { deltaDays: result, hour: 0, minute: 0, second: 0, millisecond: 0, microsecond: 0, nanosecond: 0 };
-      case 'hour':
-        return ES.BalanceTime(result, 0, 0, 0, 0, 0);
-      case 'minute':
-        return ES.BalanceTime(hour, result, 0, 0, 0, 0);
-      case 'second':
-        return ES.BalanceTime(hour, minute, result, 0, 0, 0);
-      case 'millisecond':
-        return ES.BalanceTime(hour, minute, second, result, 0, 0);
-      case 'microsecond':
-        return ES.BalanceTime(hour, minute, second, millisecond, result, 0);
-      case 'nanosecond':
-        return ES.BalanceTime(hour, minute, second, millisecond, microsecond, result);
-    }
-  },
-  DaysUntil: (earlier, later) => {
-    return ES.DifferenceISODate(
-      GetSlot(earlier, ISO_YEAR),
-      GetSlot(earlier, ISO_MONTH),
-      GetSlot(earlier, ISO_DAY),
-      GetSlot(later, ISO_YEAR),
-      GetSlot(later, ISO_MONTH),
-      GetSlot(later, ISO_DAY),
-      'day'
-    ).days;
-  },
-  MoveRelativeDate: (calendar, relativeTo, duration) => {
-    const options = ObjectCreate(null);
-    const later = ES.CalendarDateAdd(calendar, relativeTo, duration, options);
-    const days = ES.DaysUntil(relativeTo, later);
-    relativeTo = ES.CreateTemporalDateTime(
-      GetSlot(later, ISO_YEAR),
-      GetSlot(later, ISO_MONTH),
-      GetSlot(later, ISO_DAY),
-      GetSlot(relativeTo, ISO_HOUR),
-      GetSlot(relativeTo, ISO_MINUTE),
-      GetSlot(relativeTo, ISO_SECOND),
-      GetSlot(relativeTo, ISO_MILLISECOND),
-      GetSlot(relativeTo, ISO_MICROSECOND),
-      GetSlot(relativeTo, ISO_NANOSECOND),
-      GetSlot(relativeTo, CALENDAR)
-    );
-    return { relativeTo, days };
-  },
-  MoveRelativeZonedDateTime: (relativeTo, years, months, weeks, days) => {
-    const timeZone = GetSlot(relativeTo, TIME_ZONE);
-    const calendar = GetSlot(relativeTo, CALENDAR);
-    const intermediateNs = ES.AddZonedDateTime(
-      GetSlot(relativeTo, INSTANT),
-      timeZone,
-      calendar,
-      years,
-      months,
-      weeks,
-      days,
-      0,
-      0,
-      0,
-      0,
-      0,
-      0
-    );
-    return ES.CreateTemporalZonedDateTime(intermediateNs, timeZone, calendar);
-  },
-  AdjustRoundedDurationDays: (
+    dayLengthNs
+  ));
+  ({ year, month, day } = BalanceISODate(year, month, day + deltaDays));
+  return { year, month, day, hour, minute, second, millisecond, microsecond, nanosecond };
+}
+
+export function RoundTime(
+  hour,
+  minute,
+  second,
+  millisecond,
+  microsecond,
+  nanosecond,
+  increment,
+  unit,
+  roundingMode,
+  dayLengthNs = DAY_NANOS
```

Comment:
Same here: I think this should be a Number not a bigint

Reply:
Same.